lib/api/api.js

```diff
@@ -9,8 +9,26 @@ var Utils           =   require("../utils/utils");
 var async           =   require("async");
 
 'use strict';
+
 var nullHashTrytes = (new Array(244).join('9'));
 
+/**
+* @callback apiResponseCallback
+* @param {Error|String} [error]     The error object/message.  Will be `null` if the request succeeded.
+* @param {Object}       [result]    The result from the request.  Will be `null` if there was an error.
+*
+* Recommended usage:
+* ```js
+* api.doSomething(..., function(error, result) {
+*   if( error ) {
+*     // Handle error case.
+*   } else {
+*     // Handle success case.
+*   }
+* });
+* ```
+*/
+
```

Comment:
References:
  - http://usejsdoc.org/tags-param.html#callback-functions
  - http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values